### PR TITLE
Add stats and metrics

### PR DESCRIPTION
**Pull request by vmenger**
* Add (descriptive) statistics to `Dataset` class
* Add `Metrics` class which can compute some entity and qualfier metrics

### CODE REVIEW

File: src/clin_nlp_metrics/metrics.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+        Validate the two datasets. Will raise an ValueError when datasets don't contain
+        the same documents.
+        """
+        if self.true.num_docs() != self.pred.num_docs():
+            raise ValueError("Can only compute metrics for Datasets with same size")
+
+        for true_doc, pred_doc in zip(self.true.docs, self.pred.docs):
+            if true_doc.identifier != pred_doc.identifier:
+                raise ValueError(
+                    f"Found two documents with non-matching ids "
```
Comment (rubenpeters91) [inline src/clin_nlp_metrics/metrics.py:47]: I think you only need to add the f-string for the 2nd line, the others don't contain variables

File: src/clin_nlp_metrics/metrics.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+                                "doc.identifier": true_doc.identifier,
+                                "annotation": true_annotation.to_nervaluate(),
+                                "true_qualifier": true_val,
+                                "pred_qualifier": pred_val,
+                            }
+                        )
+
+        return aggregation
+
+    def qualifier_metrics(self, misses=True) -> dict:
```
Comment (rubenpeters91) [inline src/clin_nlp_metrics/metrics.py:153]: You forgot the type hint for misses

File: tests/test_metrics.py [hunk 3]
```diff
@@ -1,125 +1,123 @@
...
-            [{"start": 0, "end": 5, "label": "test1"}],
-            [{"start": 0, "end": 5, "label": "test2"}],
-        ]
+import pytest
+
+from clin_nlp_metrics import Dataset, Metrics
+
+
+@pytest.fixture
+def mctrainer_dataset():
```
Comment (rubenpeters91) [inline tests/test_metrics.py:10]: This is the same fixture as the other tests right? You can also import the fixtures

**Review by rubenpeters91 — APPROVED** [review]
Small comments, but looks good

File: README.md [hunk 4]
```diff
@@ -38,6 +40,185 @@ nlp_docs = nlp.pipe([doc['text'] for doc in data['projects'][0]['documents']])
...
+```
+
+### Metrics
+
+Create a `Metrics` object as follows:
+
+```python
+from clin_nlp_metrics import Metrics
+
+metrics = Metrics(d1, d2)
```
[the remark this answers is not in the document]
Comment (vmenger) [reply README.md:114]: Yes

File: tests/test_metrics.py (hunk 3 above)
Comment (vmenger) [reply tests/test_metrics.py]: Nice, didn't think of that. Fixed

**Pull request merged**